File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int a=0,mid;
        int b=nums.size();
        while(a<b){
            mid=a + ((b-a)/2);
            if(nums[mid]==target){
                return mid;
            }
            else if(nums[mid]<target){
                a=mid+1;
            }
            else{
                b=mid;
            }
    }
        return a;}
};

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, a: List[int], target: int) -> int:
        ans=len(a)
        low,high=0,ans-1
        while low<=high:
            mid=low+(high-low)//2
            if a[mid]>=target:
                ans=mid
                high=mid-1
            else:
                low=mid+1
        return ans